lib/elixir/lib/kernel.ex

```diff
@@ -3210,6 +3210,10 @@ defmodule Kernel do
   """
   defmacro defstruct(fields) do
     quote bind_quoted: [fields: fields] do
+      if Module.get_attribute(__MODULE__, :struct) do
+        raise ArgumentError, "defstruct has already been called for " <>
+          "#{Macro.to_string(__MODULE__)}, defstruct can only be called once per module"
```

Comment:
I'm not at my computer just now, but I think Macro.to_string prints
"MyModule". Using string interpolation prints "Elixir.MyModule".
On 4 Jan 2016 19:31, "Aleksei Magusev" <notifications@github.com> wrote:

> In lib/elixir/lib/kernel.ex
> <https://github.com/elixir-lang/elixir/pull/4156#discussion_r48769725>:
>
> > @@ -3210,6 +3210,10 @@ defmodule Kernel do
> >    """
> >    defmacro defstruct(fields) do
> >      quote bind_quoted: [fields: fields] do
> > +      if Module.get_attribute(__MODULE__, :struct) do
> > +        raise ArgumentError, "defstruct has already been called for " <>
> > +          "#{Macro.to_string(__MODULE__)}, defstruct can only be called once per module"
>
> I think there is not need for Macro.to_string/1 call.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elixir-lang/elixir/pull/4156/files#r48769725>.
>